iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer.hpp

```diff
@@ -64,11 +64,13 @@ class RelativePointer : public BaseRelativePointer
     template <typename U = T>
     typename std::enable_if<!std::is_void<U>::value, const U&>::type operator*() const noexcept;
 
-    /// @brief read-only access to the underlying object
+    /// @brief read-only access to the underlying object. If the RelativePointer does not point to anything the
+    /// application might terminate.
     /// @return a pointer to the underlying object
     T* operator->() const noexcept;
 
-    /// @brief access the underlying object
+    /// @brief access the underlying object. If the RelativePointer does not point to anything the
+    /// application might terminate.
```

Comment:
I think id **does terminate** here currently for a logical nullptr, so we should state this. Note that I think it is not a good idea to do so, because just getting the pointer (as `nullptr`)) is not the problem. The problem is dereferencing it later, as with regular pointers so we should remove the check.

Reply:
@MatthiasKillat True, I moved the assert-style check to `operator->()` and adapted the wording.